File: code/aee034b3-dc14-4581-b6b6-0bff08a6c814/eee84d2a-1d0a-4b30-929e-c3ea944caf74-4e68762e46/smoke_game.py
from pathlib import Path
root = Path(__file__).resolve().parents[1]
index = (root / 'index.html').read_text(encoding='utf-8', errors='ignore').lower()
game_path = root / 'src' / 'game.js'
assert game_path.exists(), 'missing src/game.js'
game = game_path.read_text(encoding='utf-8', errors='ignore').lower()
src_corpus = '\n'.join(p.read_text(encoding='utf-8', errors='ignore').lower() for p in (root / 'src').glob('*.js'))
corpus = index + '\n' + src_corpus
assert 'type="module"' in index or "type='module'" in index, 'index.html must load an esm module'
assert 'src/game.js' in index, 'index.html must load src/game.js'
has_progression = any(token in src_corpus for token in ('requestanimationframe', 'setinterval(', 'update(', 'tick(', 'step(', 'render(', 'renderboard', 'onsquareclick', 'movepiece', 'makemove', '.move('))
assert has_progression, 'missing game progression loop or turn handler'
assert 'keydown' in src_corpus or 'keyup' in src_corpus or 'pointerdown' in src_corpus or 'click' in src_corpus or 'input' in src_corpus, 'missing input handling'
assert 'canvas' in corpus or 'board' in corpus, 'missing primary game surface'
assert 'won' in src_corpus or 'lost' in src_corpus or 'gameover' in src_corpus or 'status' in src_corpus or 'turn' in src_corpus or 'score' in src_corpus or 'health' in src_corpus, 'missing game state'
assert 'require(' not in game and 'module.exports' not in game, 'commonjs is not allowed in browser esm stack'
assert (root / 'task_impl.py').exists() is False
assert (root / 'smoke_test.py').exists() is False
print('SMOKE_GAME_OK')

File: code/ed3b3819-817d-4b6e-b005-cc7e46005197/b01243dd-f644-4dd8-a94c-b4b054d5cc35-727c82fe2f/tests/smoke_game.mjs
import { readFileSync, existsSync } from 'node:fs';
import { dirname, join, normalize } from 'node:path';
import { fileURLToPath } from 'node:url';

const root = normalize(join(dirname(fileURLToPath(import.meta.url)), '..'));
const read = (rel) => readFileSync(join(root, rel), 'utf8');
const indexRaw = read('index.html');
const index = indexRaw.toLowerCase();
const scripts = [...indexRaw.matchAll(/<script\b([^>]*)>/gis)];
let moduleSrc = '';
for (const [, attrs] of scripts) {
  if (/\btype\s*=\s*['"]module['"]/i.test(attrs)) {
    const match = attrs.match(/\bsrc\s*=\s*['"]([^'"]+)['"]/i);
    if (match) {
      moduleSrc = match[1].split('?')[0].split('#')[0].trim().replace(/^\/+/, '');
      break;
    }
  }
}
if (!moduleSrc) throw new Error('index.html must load an esm module script with src');
if (!moduleSrc.startsWith('src/') || moduleSrc.split('/').includes('..')) throw new Error('browser entrypoint must be under src/');
if (!existsSync(join(root, moduleSrc))) throw new Error(`missing browser entrypoint ${moduleSrc}`);
const entry = read(moduleSrc);
const srcCorpusRaw = readFileSync(join(root, moduleSrc), 'utf8');
const srcCorpus = srcCorpusRaw.toLowerCase();
const corpus = `${index}\n${srcCorpus}`;
if (!/type\s*=\s*['"]module['"]/i.test(indexRaw)) throw new Error('index.html must load an esm module');
const htmlIds = new Set([...indexRaw.matchAll(/\bid\s*=\s*['"]([^'"]+)['"]/gi)].map((m) => m[1]));
const referencedIds = new Set([...srcCorpusRaw.matchAll(/getElementById\(\s*['"]([^'"]+)['"]\s*\)/g)].map((m) => m[1]));
for (const match of srcCorpusRaw.matchAll(/querySelector\(\s*['"]#([A-Za-z_][\w:-]*)['"]\s*\)/g)) referencedIds.add(match[1]);
const missingIds = [...referencedIds].filter((id) => id && !htmlIds.has(id)).sort();
if (missingIds.length) throw new Error(`entrypoint references missing DOM id(s): ${missingIds.join(',')}`);
const hasProgression = ['requestanimationframe', 'setinterval(', 'update(', 'tick(', 'step(', 'render(', 'renderboard', 'onsquareclick', 'movepiece', 'makemove', '.move('].some((token) => srcCorpus.includes(token));
if (!hasProgression) throw new Error('missing game progression loop or turn handler');
if (!['keydown', 'keyup', 'pointerdown', 'click', 'input'].some((token) => srcCorpus.includes(token))) throw new Error('missing input handling');
if (!corpus.includes('canvas') && !corpus.includes('board')) throw new Error('missing primary game surface');
if (!['won', 'lost', 'gameover', 'status', 'turn', 'score', 'health'].some((token) => srcCorpus.includes(token))) throw new Error('missing game state');
if (entry.includes('require(') || entry.includes('module.exports')) throw new Error('commonjs is not allowed in browser esm stack');
if (existsSync(join(root, 'task_impl.py'))) throw new Error('task_impl.py must not exist in browser stack');
if (existsSync(join(root, 'smoke_test.py'))) throw new Error('smoke_test.py must not exist in browser stack');
console.log('SMOKE_GAME_OK');
